miqa/core/tests/test_rest.py

```diff
@@ -1,92 +1,143 @@
+from guardian.shortcuts import get_perms
 import pytest
 
-from .fuzzy import PATH_RE
+from miqa.core.models import Project
 
 
 @pytest.mark.django_db
-def test_projects_list(authenticated_api_client, project):
-    resp = authenticated_api_client.get('/api/v1/projects')
+def test_projects_list(user_api_client, project, user):
+    resp = user_api_client().get('/api/v1/projects')
     assert resp.status_code == 200
-    assert resp.data == {
-        'count': 1,
-        'next': None,
-        'previous': None,
-        'results': [{'id': project.id, 'name': project.name}],
-    }
+    if get_perms(user, project) == []:
+        assert resp.data == {
+            'count': 0,
+            'next': None,
+            'previous': None,
+            'results': [],
+        }
+    else:
+        assert resp.data == {
+            'count': 1,
+            'next': None,
+            'previous': None,
+            'results': [{'id': project.id, 'name': project.name}],
+        }
 
 
 @pytest.mark.django_db
-def test_project_settings_get(staff_api_client, project):
-    resp = staff_api_client.get(f'/api/v1/projects/{project.id}/settings')
-    assert resp.status_code == 200
-    assert resp.data == {
-        'importPath': PATH_RE,
-        'exportPath': PATH_RE,
-    }
+def test_project_settings_get(user_api_client, project, user):
+    resp = user_api_client().get(f'/api/v1/projects/{project.id}/settings')
+    if get_perms(user, project) == []:
+        assert resp.status_code == 401
+    else:
+        assert resp.status_code == 200
+        assert all(key in resp.data for key in ['importPath', 'exportPath', 'permissions'])
 
 
 @pytest.mark.django_db
-def test_project_settings_put(staff_api_client, project):
-    staff_api_client.put(
+def test_project_settings_put(user_api_client, project, user):
+    user_api_client = user_api_client()
+    resp = user_api_client.put(
         f'/api/v1/projects/{project.id}/settings',
-        data={'importPath': '/new/fake/path', 'exportPath': '/new/fake/path'},
+        data={
+            'importPath': '/new/fake/path',
+            'exportPath': '/new/fake/path',
+            'permissions': {
+                'collaborator': [],
+                'tier_1_reviewer': [user.username],
+                'tier_2_reviewer': [],
```

Comment:
These seem like a good candidate for fixturization. Probably something like
```
@pytest.mark.fixture(
    'role',
    ['collaborator', 'tier_1_reviewer', 'tier_2_reviewer']
)
```
```suggestion
                'collaborator': [user.username] if role == 'collaborator' else [],
                'tier_1_reviewer': [user.username] if role == 'tier_1_reviewer' else [],
                'tier_2_reviewer': [user.username] if role == 'tier_2_reviewer' else [],
```

Reply:
Sorry, I don't understand where the first snippet goes to define role for the test function. Where should I put that?

Reply:
I was able to use a simpler `if` from the information we already had in the function, like this:

```
my_perms = get_perms(user, project)
expected_perms = {
        'collaborator': [user.username] if 'collaborator' in my_perms else [],
        'tier_1_reviewer': [user.username] if 'tier_1_reviewer' in my_perms else [],
        'tier_2_reviewer': [user.username] if 'tier_2_reviewer' in my_perms else [],
    }
```